packages/app-cli/app/command-mkbook.ts

```diff
@@ -0,0 +1,69 @@
+const BaseCommand = require('./base-command').default;
+const { app } = require('./app.js');
+import { _ } from '@joplin/lib/locale';
+import BaseModel from '@joplin/lib/BaseModel';
+import Folder from '@joplin/lib/models/Folder';
+import { FolderEntity } from '@joplin/lib/services/database/types';
+
+class Command extends BaseCommand {
+
+	usage() {
+		return 'mkbook <new-notebook> [notebook]';
+	}
+
+	description() {
+		return _('Creates a new notebook.');
+	}
+
+	options() {
+		return [
+			['-s, --sub', _('Creates the new notebook as a sub-notebook')],
+		];
+	}
```

Comment:
This interface is confusing. Isn't `[notebook]` required if `--sub` is present? In that case, why do we even need this additional flag? Or it should be `mkbook <new-notebook> --parent [notebook]` or just `mkbook <new-notebook> [notebook]`

Reply:
I have fix the logic, you are right the logic was confusing - my bad, thanks for review!

Reply:
Was this changed? Because it's still not good. `[notebook]` is a parameter of `-s` yet they are defined in different places. Please make it `mkbook <new-notebook>` and `-p, --parent <parent-notebook>`

Reply:
In the previous commit it was the case but currently the ` [notebook] ` is not a parameter of `-s` and is used to specify the parent folder as an optional destination.
I find your suggestion convincing, with just `mkbook <new-notebook> [notebook]`.

Reply:
I'm suggesting`-p, --parent <parent-notebook>` because it's self-documenting. For sure we don't want this trick with "--sub" and optional `[notebook]` which is not optional when "--sub" is specified. That's poor design and not intuitive.

So please make it `-p, --parent <parent-notebook>` instead. I realise it means you need to tweak your code, but once it's implemented it's pretty much forever, so it's good to get it right now.

Reply:
Understood, a question of concretization. To create a sub-notebook (test1.1) of the currently selected notebook (test1), the call looks like: "mkbook test1.1 -p test1" or "mkbook test1.1 -p $b".  I always prefer to omit unnecessary things, in this case "$b". From your experience, how would you see it if the two calls "mkbook test1.1 -p $b" and "mkbook test1.1 -p" would do the same for the currently selected notebook?

Reply:
That's a good point, it's true that we have the concept of current notebook. But so far when the notebook is implied, it's always to replace a positional parameter, which seems reasonable. But to replace a flag argument doesn't feel like a good idea, so I'd prefer if it's required in this particular case

Reply:
I hope so it should fit now - can you please check against. Thanks in advance.